modules/EvseManager/EvseManager.cpp

```diff
@@ -394,6 +394,19 @@ void EvseManager::ready() {
                     v.dc_ev_target_voltage = ev.maximum_voltage_limit.value();
                 }
 
+                bool car_breaks_limit{false};
+                const auto hlc_limits = charger->get_evse_max_hlc_limits();
+                if (v.dc_ev_target_current > hlc_limits.evse_maximum_current_limit) {
+                    v.dc_ev_target_current = hlc_limits.evse_maximum_current_limit;
+                    car_breaks_limit = true;
+                }
+
+                const auto target_power = v.dc_ev_target_current * v.dc_ev_target_voltage;
+                if (target_power > hlc_limits.evse_maximum_power_limit) {
```

Comment:
This will reduce the actual charging power quite a bit on some cars. I think we should use the actual voltage rather than the target voltage to calculate the power limit. This is what we do in other places in the code as well. If e.g. the target voltage is 420V (battery full) the actual voltage may just be 360v or so since it is empty. Then the current would be limited such that the power close to full is not higher than the max, which means it is significantly lower than the max power in the beginning of the charging session.

Reply:
You are right, I will change target_voltage to the actual voltage